cmd/mistryd/job.go

```diff
@@ -233,6 +237,35 @@ func (j *Job) StartContainer(ctx context.Context, cfg *Config, c *docker.Client,
 	return result.State.ExitCode, nil
 }
 
+// renameIfExists searches for containers with the passed name and renames them
+// by appending a random suffix to their name
+func renameIfExists(ctx context.Context, c *docker.Client, name string) error {
+	filter := filters.NewArgs()
+	filter.Add("name", name)
+	containers, err := c.ContainerList(ctx, dockertypes.ContainerListOptions{
+		Quiet:   true,
+		All:     true,
+		Limit:   -1,
+		Filters: filter,
+	})
+	if err != nil {
+		return err
+	}
+	for _, container := range containers {
+		err := c.ContainerRename(ctx, container.ID, name+"-renamed-"+randomHexString())
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func randomHexString() string {
+	buf := make([]byte, 16)
+	rand.Read(buf)
```

Comment:
Error check?

Reply:
Keep in mind that the default source you're using here...
> [...] produces a deterministic sequence of values each time a program is run.

(source: https://golang.org/pkg/math/rand)

Reply:
rand.Read can never fail, not sure why they decided to return an error, maybe to keep the same API with other Read functions?

Reply:
Ok, so if we restart the service, the same sequence of "random" values will be produced. I think this is OK. I could also use crypto/rand Read()

Reply:
There's also this recipe `rand.Seed( time.Now().UTC().UnixNano())`

Reply:
This seems like a decent choice. You only need to do this once when the server boots though.

Reply:
I'm initializing on mistry main() now, so any usage of rand called from the proper entrypoint (i.e. not from the tests) will be properly seeded